enhancements/network/external-lb-vips.md

```diff
@@ -0,0 +1,298 @@
+This will allow the user to express their desire to change the `PlatformSpec`, but the cluster could then reject it thanks to the validation.
+So, to allow that option in the future, if we start with a `PlatformStatus` field, we can add the `PlatformSpec` field later.
+If we started with a `PlatformSpec` field, moving an unknown number of consumers from `PlatformSpec` to `PlatformStatus` isn't easily possible.
+Some of that may seem like it doesn't apply for this particular use-case, but we made that decision to keep the advice consistent for all these
+fields that will be added in the future.
+
+This field will be exposed to the user in the `install-config.yaml`:
+```yaml
+platform:
+  vsphere:
+    loadBalancer:
+      type: UserManaged
+featureSet: TechPreviewNoUpgrade
+```
+
+If the user wants to be explicit and set the default load balancer which is OpenShift managed:
+```yaml
+platform:
+  vsphere:
+    loadBalancer:
+      type: OpenShiftManagedDefault
+featureSet: TechPreviewNoUpgrade
+```
+
+If the user deploys with `featureSet: TechPreviewNoUpgrade` but doesn't set the `loadBalancer` field, the default load balancer will be deployed for backward compatibility.
+
+Note that `loadBalancer` is immutable and once set can't be changed for now.
+In the future, we'll be able to add more load balancers and even customize them by adding more options.
+
+The following platforms will have this feature as TechPreview:
+* BareMetal
+* Nutanix
+* OpenStack
+* Ovirt
+* vSphere
+
+### Implementation Details/Notes/Constraints
+
+* CoreDNS will be deployed if external loadbalancer is being used.
+  However, it is not yet supported to change the DNS records for the API and Ingress VIPs in case the
+  IPs have changed. This will be supported in the [future](https://issues.redhat.com/browse/RFE-2085).
+
+
+* We need to make it ready for any platform who wants to support it, and allow
+  platforms to turn this on in the installer project. This will be achieved while validating the
+  installconfig manifest.
+
+* In the case of a control-plane replacement, the customer has to make sure that
+  the loadbalancer backends already have the IP address of the node that will be created during
+  the replacement. Since the cluster doesn't host the VIPs nor the loadbalancer, there is nothing
+  else to care now.
+  However in the future with Static IPs, we'll probably have a new controller for IPAM but this is out
+  of scope now.
+
+* For `vsphere` platform, since they're already have a way to disable the internal loadbalancer by
+  not specifying the VIPs in `PlatformStatus`, we'll keep this code in place but only for this platform
+  so backward compatibility is maintained.
+
+### Risks and Mitigations
+
+Some platforms (vsphere, openstack) have been testing it already and some issues have been identified:
+* Having an external LB breaks [nodeip-finder](https://github.com/openshift/baremetal-runtimecfg/blob/master/scripts/nodeip-finder) script.
```

Comment:
Ideally we would not reference nodeip-finder. That script hasn't been used in ages. All of the logic now lives in the "runtimecfg node-ip [show|set]" logic.

Reply:
ack. I'll remove it the next time I have to edit the file, in this PR or in a follow-up.